measurements/toxicity/toxicity.py

```diff
@@ -0,0 +1,146 @@
+        >>> model.config.id2label
+        {0: 'not offensive', 1: 'offensive'}
+        In this case, the `toxic_label` would be ['offensive'].
+    `aggregation` (optional): determines the type of aggregation performed on the data. If set to `None`, the scores for each prediction are returned.
+     Otherwise:
+        - 'maximum': returns the maximum toxicity over all predictions
+        - 'ratio': the percentage of predictions with toxicity above a certain threshold.
+    `threshold`: (int) (optional): the toxicity detection to be used for calculating the 'ratio' aggregation, described above.
+    The default threshold is 0.5, based on the one established by [RealToxicityPrompts](https://arxiv.org/abs/2009.11462).
+
+Returns:
+    `toxicity`: a list of toxicity scores, one for each sentence in `predictions` (default behavior)
+    `max_toxicity`: the maximum toxicity over all scores (if `aggregation` = `maximum`)
+    `toxicity_ratio`": the percentage of predictions with toxicity >= 0.5 (if `aggregation` = `ratio`)
+
+Examples:
+
+    Example 1 (default behavior):
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts)
+        >>> print(results)
+        {'toxicity': [0.0002070277841994539, 0.856372594833374]}
+
+    Example 2 (returns ratio of toxic sentences):
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts, aggregation = "ratio")
+        >>> print(results)
+        {'toxicity_ratio': 0.5}
+
+    Example 3 (returns the maximum toxicity score):
+
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts, aggregation = "maximum")
+        >>> print(results)
+        {'max_toxicity': 0.856372594833374}
+
+    Example 4 (uses a custom model):
+
+        >>> toxicity = evaluate.load("toxicity", 'DaNLP/da-electra-hatespeech-detection')
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts, toxic_label='offensive')
+        >>> print(results)
+        {'toxicity': [0.017554517835378647, 0.020320769399404526]}
+"""
+
+
+def toxicity(preds, toxic_classifier, toxic_label):
+    toxic_scores = []
+    if toxic_label in toxic_classifier.model.config.id2label.values():
+        for pred in preds:
+            pred_toxic = toxic_classifier(str(pred))
+            hate_toxic = [r["score"] for r in pred_toxic if r["label"] == toxic_label][0]
+            toxic_scores.append(hate_toxic)
```

Comment:
You can actually pass all the predictions at once:

```python
for pred_toxic in toxic_classifier(preds):
    hate_toxic = [r["score"] for r in pred_toxic if r["label"] == toxic_label][0]
```

Why do you need to cast the preds to `str`? They should be strings, no?